ui/app/adapters/identity/oidc/config.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import ApplicationAdapter from '../../application';
+
+export default class IdentityOidcConfig extends ApplicationAdapter {
+  namespace = 'v1';
+
+  queryRecord() {
+    return this.ajax(`${this.buildURL()}/identity/oidc/config`, 'GET').then((resp) => {
+      return {
+        ...resp,
+        id: resp.request_id, // id required for ember data
+      };
+    });
+  }
+
+  createOrUpdate(store, type, snapshot) {
+    const serializer = store.serializerFor(type.modelName);
+    const data = serializer.serialize(snapshot);
+    if (!data.id) data.id = data.issuer;
```

Comment:
Is `id` required for a POST request? It seems like we don't need this line since the API generates an id for us.

Reply:
It's a testing thing. I can leave a comment or attack tmrw with a fresh brain why it kept failing. But it would say request needs id.